File: WebMobile/modele/Utilisateur.php
<?php
// Projet TraceGPS
// fichier : modele/Utilisateur.php
// Rôle : la classe Utilisateur représente les utilisateurs de l'application
// Dernière mise à jour : 9/7/2021 par dP
include_once ('Outils.php');
class Utilisateur
{
    // ------------------------------------------------------------------------------------------------------
    // ---------------------------------- Attributs privés de la classe -------------------------------------
    // ------------------------------------------------------------------------------------------------------
    private $id; // identifiant de l'utilisateur (numéro automatique dans la BDD)
    private $pseudo; // pseudo de l'utilisateur
    private $mdpSha1; // mot de passe de l'utilisateur (hashé en SHA1)
    private $adrMail; // adresse mail de l'utilisateur
    private $numTel; // numéro de téléphone de l'utilisateur
    private $niveau; // niveau d'accès : 1 = utilisateur (pratiquant ou proche) 2 = administrateur
    private $dateCreation; // date de création du compte
    private $nbTraces; // nombre de traces stockées actuellement
    private $dateDerniereTrace; // date de début de la dernière trace
    
    // ------------------------------------------------------------------------------------------------------
    // ----------------------------------------- Constructeur -----------------------------------------------
    // ------------------------------------------------------------------------------------------------------
    public function __construct($unId, $unPseudo, $unMdpSha1, $uneAdrMail, $unNumTel, $unNiveau,
        $uneDateCreation, $unNbTraces, $uneDateDerniereTrace) {
            $this->id = $unId;
            $this->pseudo = $unPseudo;
            $this->mdpSha1 = $unMdpSha1;
            $this->adrMail = $uneAdrMail;
            $this->numTel = Outils::corrigerTelephone($unNumTel);
            $this->niveau = $unNiveau;
            $this->dateCreation = $uneDateCreation;
            $this->nbTraces = $unNbTraces;
            $this->dateDerniereTrace = $uneDateDerniereTrace;
    }
    
    // ------------------------------------------------------------------------------------------------------
    // ---------------------------------------- Getters et Setters ------------------------------------------
    // ------------------------------------------------------------------------------------------------------
    public function getId() {return $this->id;}
    public function setId($unId) { $this->id = $unId;}
    
    public function getPseudo() {return $this->pseudo;}
    public function setPseudo($unPseudo) {$this->pseudo = $unPseudo;}
    
    public function getMdpSha1() {return $this->mdpSha1;}
    public function setMdpSha1($unMdpSha1) {$this->mdpSha1 = $unMdpSha1;}
    
    public function getAdrMail() {return $this->adrMail;}
    public function setAdrMail($uneAdrMail) {$this->adrMail = $uneAdrMail;}
    
    public function getNumTel() {return $this->numTel;}
    public function setNumTel($unNumTel) {$this->numTel = Outils::corrigerTelephone($unNumTel);}
    
    public function getNiveau() {return $this->niveau;}    
    public function setNiveau($unNiveau) {$this->niveau = $unNiveau;}
    
    public function getDateCreation() {return $this->dateCreation;}    
    public function setDateCreation($uneDateCreation) {$this->dateCreation = $uneDateCreation;}
    
    public function getNbTraces() {return $this->nbTraces;}    
    public function setNbTraces($unNbTraces) {$this->nbTraces = $unNbTraces;}
    
    public function getDateDerniereTrace() {return $this->dateDerniereTrace;}
    public function setDateDerniereTrace($uneDateDerniereTrace) {$this->dateDerniereTrace = $uneDateDerniereTrace;}
    
    // ------------------------------------------------------------------------------------------------------
    // -------------------------------------- Méthodes d'instances ------------------------------------------
    // ------------------------------------------------------------------------------------------------------
    public function toString() {
        $msg = 'id : ' . $this->id . '<br>';
        $msg .= 'pseudo : ' . $this->pseudo . '<br>';
        $msg .= 'mdpSha1 : ' . $this->mdpSha1 . '<br>';
        $msg .= 'adrMail : ' . $this->adrMail . '<br>';
        $msg .= 'numTel : ' . $this->numTel . '<br>';
        $msg .= 'niveau : ' . $this->niveau . '<br>';
        $msg .= 'dateCreation : ' . $this->dateCreation . '<br>';
        $msg .= 'nbTraces : ' . $this->nbTraces . '<br>';
        $msg .= 'dateDerniereTrace : ' . $this->dateDerniereTrace . '<br>';
        return $msg;
    }
    
    
} // fin de la classe Utilisateur

File: C#/TraceGPS_C#_fourni/TraceGPS/modele/Utilisateur.cs
// Projet TraceGPS
// fichier : modele/Point.cs
// Rôle : la classe Point représente un point géographique
// Dernière mise à jour : 1/11/2021 par dp

using System;

namespace TraceGPS
{
    public class Utilisateur
    {
        // attributs privés ---------------------------------------------------------------------------

        private int _id;					// identifiant de l'utilisateur (numéro automatique dans la BDD)
        private String _pseudo;				// pseudo de l'utilisateur
        private String _mdpSha1;			// mot de passe de l'utilisateur (hashé en SHA1)
        private String _adrMail;			// adresse mail de l'utilisateur
        private String _numTel;				// numéro de téléphone de l'utilisateur
        private int _niveau;				// niveau d'accès : 1 = utilisateur (pratiquant ou proche)    2 = administrateur
        private DateTime _dateCreation;		// date de création du compte
        private int _nbTraces;				// nombre de traces stockées actuellement
        private DateTime _dateDerniereTrace;// date de début de la dernière trace	

        // Constructeur sans paramètre
        public Utilisateur()
        {
            _id = 0;
            _pseudo = "";
            _mdpSha1 = "";
            _adrMail = "";
            _numTel = "";
            _niveau = 0;
            _dateCreation = DateTime.MinValue;
            _nbTraces = 0;
            _dateDerniereTrace = DateTime.MinValue;
        }

        // Constructeur avec paramètres
        public Utilisateur(int unId, String unPseudo, String unMdpSha1, String uneAdrMail, String unNumTel,
                int unNiveau, DateTime uneDateCreation, int unNbTraces, DateTime uneDateDerniereTrace)
        {
            _id = unId;
            _pseudo = unPseudo;
            _mdpSha1 = unMdpSha1;
            _adrMail = uneAdrMail;
            _numTel = Outils.corrigerTelephone(unNumTel);
            _niveau = unNiveau;
            _dateCreation = uneDateCreation;
            _nbTraces = unNbTraces;
            _dateDerniereTrace = uneDateDerniereTrace;
        }

        // Accesseurs ---------------------------------------------------------------------------------

        public int getId() { return _id; }
        public void setId(int unId) { this._id = unId; }

        public String getPseudo() { return _pseudo; }
        public void setPseudo(String unPseudo) { _pseudo = unPseudo; }

        public String getMdpSha1() { return _mdpSha1; }
        public void setMdpSha1(String unMdpSha1) { _mdpSha1 = unMdpSha1; }

        public String getAdrMail() { return _adrMail; }
        public void setAdrMail(String uneAdrMail) { _adrMail = uneAdrMail; }

        public String getNumTel() { return _numTel; }
        public void setNumTel(String unNumTel) { _numTel = Outils.corrigerTelephone(unNumTel); }

        public int getNiveau() { return _niveau; }
        public void setNiveau(int unNiveau) { _niveau = unNiveau; }

        public DateTime getDateCreation() { return _dateCreation; }
        public void setDateCreation(DateTime uneDateCreation) { _dateCreation = uneDateCreation; }

        public int getNbTraces() { return _nbTraces; }
        public void setNbTraces(int unNbTraces) { _nbTraces = unNbTraces; }

        public DateTime getDateDerniereTrace() { return _dateDerniereTrace; }
        public void setDateDerniereTrace(DateTime uneDateDerniereTrace) { _dateDerniereTrace = uneDateDerniereTrace; }

        // Méthodes publiques -------------------------------------------------------------------------

        // Fournit une chaine contenant toutes les données de l'objet
        public String toString()
        {
            String msg = "";
            msg += "id : " + _id + "\n";
            msg += "pseudo : " + _pseudo + "\n";
            msg += "mdpSha1 : " + _mdpSha1 + "\n";
            msg += "adrMail : " + _adrMail + "\n";
            msg += "numTel : " + _numTel + "\n";
            msg += "niveau : " + _niveau + "\n";
            if (this._dateCreation != null)
                msg += "dateCreation : " + _dateCreation.ToString("dd/MM/yyyy HH:mm:ss") + "\n";
            msg += "nbTraces : " + _nbTraces + "\n";
            if (_nbTraces > 0)
                msg += "dateDerniereTrace : " + _dateDerniereTrace.ToString("dd/MM/yyyy HH:mm:ss") + "\n";

            return msg;
        }
    }
}
